src/evo/netinfo.h

```diff
@@ -60,6 +60,35 @@ constexpr std::string_view NISToString(const NetInfoStatus code)
     assert(false);
 }
 
+// Purpose corresponds to the index position in the ExtNetInfo map (which is ordered),
+// entries must be contiguous and cannot be changed once set without a format version
+// update, which will necessitate a hard-fork.
+namespace Purpose {
```

Comment:
nit: this name is way to general for using in global namespace; consider using here instead of just Purpose something like NetPurpose or `AddressPurpose` or NetInfoPurpose` or something other less generic